File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/PersonRequest.cpp
#include <string>
#include <optional>
#include <algorithm>
#include <cctype>

class PersonRequest {
public:
    std::optional<std::string> name;
    std::optional<std::string> sex;
    std::optional<std::string> phoneNumber;

    PersonRequest(const std::string& name, const std::string& sex, const std::string& phoneNumber)
        : name(validate_name(name)),
          sex(validate_sex(sex)),
          phoneNumber(validate_phoneNumber(phoneNumber)) {}

private:
    static std::optional<std::string> validate_name(const std::string& name) {
        if (name.empty()) return std::nullopt;
        if (name.size() > 33) return std::nullopt;
        return name;
    }

    static std::optional<std::string> validate_sex(const std::string& sex) {
        if (sex != "Man" && sex != "Woman" && sex != "UGM") return std::nullopt;
        return sex;
    }

    static std::optional<std::string> validate_phoneNumber(const std::string& phoneNumber) {
        if (phoneNumber.empty()) return std::nullopt;
        if (phoneNumber.size() != 11) return std::nullopt;
        if (!std::all_of(phoneNumber.begin(), phoneNumber.end(),
                         [](unsigned char c) { return std::isdigit(c); }))
            return std::nullopt;
        return phoneNumber;
    }
};

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/cpp_to_py/PersonRequest.py
class PersonRequest:

    def __init__(self, name, sex, phoneNumber):
        self.name = self._validate_name(name)
        self.sex = self._validate_sex(sex)
        self.phoneNumber = self._validate_phoneNumber(phoneNumber)

    def _validate_name(self, name):
        if name == '' or len(name) > 33:
            return ''
        return name

    def _validate_sex(self, sex):
        if sex not in ['Man', 'Woman', 'UGM']:
            return ''
        return sex

    def _validate_phoneNumber(self, phoneNumber):
        if phoneNumber == '' or len(phoneNumber) != 11 or (not self._is_all_digits(phoneNumber)):
            return ''
        return phoneNumber

    def _is_all_digits(self, s):
        for c in s:
            if c not in '0123456789':
                return False
        return True